lib/service/service.go

```diff
@@ -3299,6 +3332,31 @@ func proxySettingsFromConfig(cfg *Config, proxySSHAddr utils.NetAddr) webclient.
 	return proxySettings
 }
 
+// getPostgresPublicAddr returns the proxy PostgresPublicAddrs based if proxy postgres service
+// was configured on separate listener. For backward compatibility if PostgresPublicAddrs was not provided.
+// Proxy will reuse the PostgresPublicAddrs field to propagate postgres service address to legacy tsh clients.
+func getPostgresPublicAddr(cfg *Config) string {
+	if len(cfg.Proxy.PostgresPublicAddrs) > 0 {
+		return cfg.Proxy.PostgresPublicAddrs[0].String()
+	}
+
+	if len(cfg.Proxy.PostgresPublicAddrs) != 0 || cfg.Proxy.PostgresAddr.IsEmpty() {
```

Comment:
Is this condition correct? It seems like `len(cfg.Proxy.PostgresPublicAddrs) != 0` would already be handled by the `> 0` check above so it would never be true here?

Reply:
fixed